pyccl/core.py

```diff
@@ -577,73 +569,80 @@ def _build_parameters(
-                Omega_c, Omega_b, Omega_k, Neff, w0, wa, h, A_s, sigma8, n_s,
-                T_CMB, Omega_g, T_ncdm, bcm_log10Mc, bcm_etab, bcm_ks,
-                mu_0, sigma_0, c1_mg, c2_mg, lambda_mg, z_mg, df_mg,
-                mnu_final_list, status)
-        check(status)
+            # Omega_g was passed - modify Omega_l
+            Omega_l += rho_g/rho_crit - Omega_g
+
+        self._fill_params(
+            sum_nu_masses=sum(nu_mass), N_nu_rel=N_nu_rel, Neff=Neff,
+            Omega_nu_mass=Omega_nu_mass, Omega_nu_rel=Omega_nu_rel,
+            Omega_m=Omega_m, Omega_c=Omega_c, Omega_b=Omega_b, Omega_k=Omega_k,
+            sqrtk=sqrtk, k_sign=int(k_sign), Omega_g=Omega_g, w0=w0, wa=wa,
+            Omega_l=Omega_l, H0=h*100, A_s=A_s, sigma8=sigma8, n_s=n_s,
+            mu_0=mu_0, sigma_0=sigma_0, c1_mg=c1_mg, c2_mg=c2_mg,
+            lambda_mg=lambda_mg,
+            bcm_log10Mc=bcm_log10Mc, bcm_etab=bcm_etab, bcm_ks=bcm_ks)
+
+        # Modified growth (deprecated)
+        if z_mg is not None:
+            self._params.mgrowth = [z_mg, df_mg]
+
+    def _fill_params(self, **kwargs):
+        if not hasattr(self, "_params"):
+            self._params = CosmologyParams()
+        [setattr(self._params, par, val) for par, val in kwargs.items()]
 
     def __getitem__(self, key):
-        """Access parameter values by name."""
-        try:
-            if key == 'm_nu':
-                val = lib.parameters_get_nu_masses(self._params, 3)
-            elif key == 'extra_parameters':
-                val = self._params_init_kwargs["extra_parameters"]
-            else:
-                val = getattr(self._params, key)
-        except AttributeError:
-            raise KeyError("Parameter '%s' not recognized." % key)
-        return val
-
-    def __setitem__(self, key, val):
-        """Set parameter values by name."""
-        raise NotImplementedError("Cosmology objects are immutable; create a "
-                                  "new Cosmology() instance instead.")
+        if key == 'extra_parameters':
+            return self._params_init_kwargs["extra_parameters"]
+        return getattr(self._params, key)
 
     def __del__(self):
         """Free the C memory this object is managing as it is being garbage
         collected (hopefully)."""
         if hasattr(self, "cosmo"):
-            if (self.cosmo is not None and
-                    hasattr(lib, 'cosmology_free') and
-                    lib.cosmology_free is not None):
-                lib.cosmology_free(self.cosmo)
-        if hasattr(self, "_params"):
-            if (self._params is not None and
-                    hasattr(lib, 'parameters_free') and
-                    lib.parameters_free is not None):
-                lib.parameters_free(self._params)
-
-        # finally delete some attributes we don't want to be around for safety
-        # when the context manager exits or if __del__ is called twice
-        if hasattr(self, "cosmo"):
```

Comment:
Why do we not need these extra checks now?

Reply:
Because the way it is set up you can't have something assigned to `cosmo.cosmo` and that something not being a SWIG object:
```python
        self.cosmo = lib.cosmology_create(self._params, self._config)
```
If this step fails, nothing is assigned to `self.cosmo`.